imageGenSD.py

```diff
@@ -355,35 +356,45 @@ async def hello(interaction: dis.Interaction):
 
 @IGSD_client.tree.command()
 @dac.checks.has_permissions(use_application_commands=True)
-@dac.describe(user="The Discord user owning the profiles lsited by the command.  If none, it defaults to you.")
 @dac.describe(name="A name, or fragment of, used to filter the profile list (case insensitive).")
+@dac.describe(rarity="Which rarity to search.  If none, if defaults to showing all rarities.")
+@dac.describe(user="The Discord user owning the profiles lsited by the command.  If none, it defaults to you.")
 async def listprofiles(interaction : dis.Interaction,
-                       user        : Optional[dis.User] = None,
-                       name        : Optional[dac.Range[str, 0, 36]] = None):
+                       name        : Optional[dac.Range[str, 0, 36]] = None,
+                       rarity      : Optional[rc.RarityList]         = None,
+                       user        : Optional[dis.User]              = None,):
     """Returns a pagenated list of profile names and IDs owned by the caller.
 
         Input  : interaction - the interaction context from Discord.
                  user - optional, the discord user whose profiles to list
+                 rarity - optional, a rarity to filter results on
                  name - optional, a name or fragment thereof to filter the profile list
 
         Output : N/A.
     """
 
-    dis_log  = log.getLogger('discord')
+    dis_log       = log.getLogger('discord')
+    error_desc    = ""
+    message_title = ""
+    profiles      = []
+    rarity_values = None if rarity == None else rarity.value
     #This has to be in the function body because an arg can't be used to assign
     #another arg in the function call.
-    user_id = interaction.user.id if user == None else user.id
-    user_dis = interaction.user if user == None else user
-    error_desc = ""
-    profiles = []
-    message_title = ""
+    user_id       = interaction.user.id if user == None else user.id
+    user_dis      = interaction.user if user == None else user
+
+    if rarity == None :
+
+        rarities      = [str(x.value) for x in rc.RarityList]
+        rarity_values = ','.join(rarities)
+
     if name == None:
-        profiles = db_ifc.getUsersProfiles(user_id)
-        error_desc = f"User <@{user_id}> does not own any characters!"
+        profiles      = db_ifc.getUsersProfiles(rarity_values, user_id)
+        error_desc    = f"User <@{user_id}> does not own any characters!"
```

Comment:
Why no custom error text. Apparently if i want to list my SUPER_RAREs, I no longer own any characters :(

Reply:
Added, with many ternaries of sadness.